torch/csrc/jit/codegen/cuda/ir_base_nodes.h

```diff
@@ -95,6 +95,29 @@ struct TORCH_CUDA_API Statement {
   // Make sure this is an Expr and return it as an Expr*
   Expr* asExpr();
 
+  // Replacement for static_cast<T*>(ptr): ptr->as<T>()
+  template <class T>
+  T* as() {
+#ifdef NDEBUG
+    auto downcast_ptr = static_cast<T*>(this);
+#else
+    auto downcast_ptr = dynamic_cast<T*>(this);
+#endif
+    assert(downcast_ptr != nullptr);
```

Comment:
Can we do an explicit RTTI check to make sure the cast is valid?

Reply:
@jjsjann123 : I moved the check inside the #else path and replaced it with TORCH_INTERNAL_ASSERT. There's a small problem the TORCH_INTERNAL_ASSERT (and TORCK_CHECK) - they report failures as exceptions. Which means that it's problematic to have anything which uses them in destructors.

> Can we do an explicit RTTI check to make sure the cast is valid?
Yes, that's the dynamic_cast<> path in debug builds (no check in release builds since this is a replacement for static_casts)